lib/string_decoder.js

```diff
@@ -72,9 +72,11 @@ StringDecoder.prototype.write = function write(buf) {
   if (typeof buf === 'string')
     return buf;
   if (!ArrayBuffer.isView(buf))
-    throw new ERR_INVALID_ARG_TYPE('buf',
-                                   ['Buffer', 'Uint8Array', 'ArrayBufferView'],
-                                   buf);
+    throw new ERR_INVALID_ARG_TYPE(
+      'buf',
+      ['Buffer', 'TypedArray', 'DataView', 'ArrayBufferView'],
```

Comment:
While we're in here, does it make sense to alphabetize the list? Or is the existing ordering logical for other reasons?

Reply:
I think the list makes sense in that `Buffer` will be the most common type of argument (by far), followed by `TypedArray`s.

I am not sure why `ArrayBufferView` is listed explicitly, since currently there are (afaik) only `TypedArray` and `DataView` ABVs.